R/transform-files.R

```diff
@@ -154,5 +154,43 @@ apply_transformers <- function(pd_nested, transformers) {
     outer_indention_refs = NA
   )
   transformed_absolute_indent
+}
 
+
+#' Verify the styling if possible
+#'
+#' If scope was set to "line_breaks" or lower (compare [tidyverse_style()]),
+#' we can compare the expression before and after styling and return an error if
+#' it is not the same. Note that this method ignores comments and no
+#' verification can be conducted if scope > "line_breaks".
+#' @param old_text The initial expression in its character representation.
+#' @param new_text The styled expression in its character representation.
+#' @param transformers The list of transformer functions used for styling.
+#'   Needed for reverse engineering the scope.
+#' @examples
+#' styler:::verify_roundtrip("a+1", "a + 1", tidyverse_style(scope = "line_breaks"))
+#' styler:::verify_roundtrip(
+#'   "a+1",
+#'   "a + 1 # comments are dropped",
+#'   tidyverse_style(scope = "line_breaks")
+#' )
+#' \dontrun{
+#' styler:::verify_roundtrip("a+1", "b - 3", tidyverse_style(scope = "line_breaks"))
+#' }
+verify_roundtrip <- function(old_text, new_text, transformers) {
+  if (is.null(transformers$token)) {
```

Comment:
I don't see a better way, but we could make this explicit by extracting this to a function:

```r
  if (can_verify_roundtrip(transformers)) {
    ...
  }
```